addons/binding/org.openhab.binding.meteostick/src/main/java/org/openhab/binding/meteostick/handler/meteostickBridgeHandler.java

```diff
@@ -0,0 +1,341 @@
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.TooManyListenersException;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+import org.eclipse.smarthome.config.core.Configuration;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandlerCallback;
+import org.eclipse.smarthome.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import gnu.io.CommPort;
+import gnu.io.CommPortIdentifier;
+import gnu.io.NoSuchPortException;
+import gnu.io.PortInUseException;
+import gnu.io.SerialPort;
+import gnu.io.SerialPortEvent;
+import gnu.io.SerialPortEventListener;
+import gnu.io.UnsupportedCommOperationException;
+
+/**
+ * The {@link meteostickBridgeHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Chris Jackson - Initial contribution
+ */
+public class meteostickBridgeHandler extends BaseThingHandler {
+    public final static Set<ThingTypeUID> SUPPORTED_THING_TYPES = Collections.singleton(THING_TYPE_BRIDGE);
+
+    private Logger logger = LoggerFactory.getLogger(meteostickBridgeHandler.class);
+
+    private static int RECEIVE_TIMEOUT = 3000;
+
+    private SerialPort serialPort;
+    private ReceiveThread receiveThread;
+
+    private String meteostickMode = "m1";
+    private String meteostickChannels = "t0";
+    private final String meteostickFormat = "o1";
+
+    private Date lastData;
+    private Timer offlineTimer = new Timer();
+    private TimerTask offlineTimerTask = null;
+
+    private ConcurrentMap<Integer, meteostickEventListener> eventListeners = new ConcurrentHashMap<Integer, meteostickEventListener>();
+
+    public meteostickBridgeHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void setCallback(ThingHandlerCallback thingHandlerCallback) {
+        logger.debug("Callback is set to: {}.", thingHandlerCallback, new RuntimeException("log stacktrace"));
+        super.setCallback(thingHandlerCallback);
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing MeteoStick Bridge handler.");
+        super.initialize();
+
+        updateStatus(ThingStatus.OFFLINE);
+
+        Configuration config = getThing().getConfiguration();
+
+        String port = (String) config.get("port");
+        connectPort(port);
+    }
+
+    @Override
+    public void dispose() {
+        disconnect();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+    }
+
+    private void createChannelCommand() {
```

Comment:
This is an ugly method with unexpected side-effects.
It would be much better to take the keyset of the eventListeners as a parameter and return a string, which is then used to set meteostickChannels.